gpkit/constraints/link.py

```diff
@@ -0,0 +1,57 @@
+"Implements LinkConstraint"
+from .set import ConstraintSet
+from ..varkey import VarKey
+from .. import SignomialsEnabled
+
+
+class LinkConstraint(ConstraintSet):
+    """A ConstraintSet that links duplicate variables in its constraints
+
+    Variables with the same name are linked
+
+    Arguments
+    ---------
+    constraints: iterable
+        valid argument to ConstraintSet
+    include_only: set
+        whitelist of variable names to include
+    exclude: set
+        blacklist of variable names, supercedes include_only
+    """
```

Comment:
@bqpd please review this docstring. It needs more info on when things are linked -- including info on having different `value`, `models`, or `modelnums` affects linking -- please add.